test/SplitCommsTest.cpp

```diff
@@ -0,0 +1,142 @@
+/*************************************************************************
+ * Copyright (c) 2023 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * See LICENSE.txt for license information
+ ************************************************************************/
+
+#include <gtest/gtest.h>
+#include <rccl/rccl.h>
+
+#include "StandaloneUtils.hpp"
+
+class SplitCommsTest : public ::testing::Test {
+protected:
+    std::vector<ncclComm_t> comms;
+    int numDevices;
+
+    void SetUp() override {
+        HIPCALL(hipGetDeviceCount(&numDevices));
+
+        if (numDevices < 2) {
+            GTEST_SKIP() << "This test requires at least 2 GPUs.";
+        }
+
+        ncclUniqueId id;
+        comms.resize(numDevices);
+
+        NCCLCHECK(ncclGetUniqueId(&id));
+
+        // Initialize the original comms
+        NCCLCHECK(ncclGroupStart());
+        for (int i = 0; i < numDevices; i++) {
+            HIPCALL(hipSetDevice(i));
+            NCCLCHECK(ncclCommInitRank(&comms[i], numDevices, id, i));
+        }
+        NCCLCHECK(ncclGroupEnd());
```

Comment:
Use ncclCommInitAll instead, to avoid all the resources required by ncclGetUniqueId